extensions/applicationinsights-angular-js/src/AngularPlugin.ts

```diff
@@ -30,19 +32,24 @@ export default class AngularPlugin extends BaseTelemetryPlugin {
             }
         });
         if (extConfig.router) {
+            let isPageInitialLoad = true;
+            if (isPageInitialLoad) {
+                const pageViewTelemetry: IPageViewTelemetry = {
+                    uri: extConfig.router.url
+                };
+                this.trackPageView(pageViewTelemetry);
+            }
             extConfig.router.events.subscribe(event => {
-                if (event.constructor.name === "NavigationEnd") {
-                    // Timeout to ensure any changes to the DOM made by route changes get included in pageView telemetry
-                    setTimeout(() => {
-                        const pageViewTelemetry: IPageViewTelemetry = { uri: extConfig.router.url };
-                        this.trackPageView(pageViewTelemetry);
-                    }, 500);
+                if (event.constructor.name === NAVIGATIONEND) {
+                    // for page initial load, do not call trackPageView twice
+                    if (isPageInitialLoad) {
+                        isPageInitialLoad = false;
+                        return;
+                    }
+                    const pageViewTelemetry: IPageViewTelemetry = { uri: extConfig.router.url };
+                    this.trackPageView(pageViewTelemetry);
```

Comment:
Update operation id before calling `trackPageView`

Reply:
oh I just pushed a commit before I saw your message. I added this, but find the properties plugin itself to update operation id. Can you please take a look if it makes sense. The `this._anaylticsPlugin['_properties']` returns `undefined`

Reply:
There may be a bug in the analytics plugin, but this is where we search for the properties plugin there
https://github.com/microsoft/ApplicationInsights-JS/blob/ab359e7d6714577800672945107b5c7b0e5d1af0/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts#L624-L628